subprojects/base-services/src/main/java/org/gradle/util/GradleVersion.java

```diff
@@ -308,4 +308,17 @@ public int compareTo(Stage other) {
             return 0;
         }
     }
+
+    public boolean isSameOrNewer(String otherVersion) {
+        return isVersion(otherVersion) || this.compareTo(GradleVersion.version(otherVersion)) > 0;
+    }
+
+    public boolean isSameOrOlder(String otherVersion) {
```

Comment:
This was actually coming from `DefaultGradleDistribution` but was moved. I didn't want to change too much of the testing infrastructure. It's now refactored and in use.